File: backend/CoffeeAPI/Models/OrderStatus.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace CoffeeAPI.Models
{
    public class OrderStatus
    {
        public Guid OrderStatusId { get; set; }
        [Required]
        public string StatusName { get; set; }

        public ICollection<OrderLine> OrderLines { get; set; }
    }
}

File: frontend/src/app/core/classes/order-status.ts
export class OrderStatus {
    orderStatusId: string;
    statusName: string;
    public static statusArray = [];

    constructor(orderStatusId: string, statusName: string) {
        this.orderStatusId = orderStatusId;
        this.statusName = statusName;
        OrderStatus.statusArray.push(this);
    }
}

